dasch-swiss/dsp-js-lib
# Create and delete ontology

Resolves [426](https://dasch.myjetbrains.com/youtrack/issue/DSP-426)

This PR uses Knora/DSP-API v13.0.0-rc.8.
Create and delete an ontology is the main part of this PR.

You can test it in test-framework:
1. Get the latest version of the branch: `git pull`
1. Install all depndencies: `npm install && npm run peer-deps`
1. Prepare and build the lib: `npm run prepare-dev-publication && npm run yalc-publish`
    (prepare-dev-publication will update 3 files. `git status` will show them as changed files. Please do not commit or push them!)
1. Go to test-framework: `cd test-framework`
1. Prepare test-framework (from dsp-js-lib test-framework folder): `yalc remove --all && yalc add @dasch-swiss/dsp-js && npm install`
1. Run the test-framework app: `ng s`
1. Open browser and go to: <http://localhost:4200>
1. Be sure that Knora/DSP-API is running in the background
1. Click on `LOGIN` button
1. Go to section "Ontology" and click the buttons in the following order:
  * `Create testonto` will create a test ontology in anything project
  * `Get testonto` will grab the data of new created test ontology
  * Optional: Click on `Get anything project ontologies` in section "Project specific ontologies" and you should get 4 instead of 3 ontologies
  * `Delete testonto` will delete the testonto
  * Optional: Click on `Get anything project ontologies` in section "Project specific ontologies" and you should get 3 ontologies again
